conformance/lang/expressions/is-expr/is_expr.balt

```diff
@@ -0,0 +1,1767 @@
+Test-Case: output
+Description: Test nil literal as expression in type test expression.
+Labels: is-expr, boolean, nil-literal, nil-type
+
+function init() {
+    boolean b = () is ();
```

Comment:
Shouldn't this be allowed?

```ballerina
true|false res = () is ();
```

Reply:
The static type of the result of the `is-expr` is `boolean` and we also don't support this for other expressions whose static type of the result of the expression is `boolean` (ex: relational-expr, logical-expr) right?

Reply:
As discussed offline, please create a fail-issue test